### PR TITLE
ci: added jest configs, added github actions, added modern only suppo…

### PR DESCRIPTION
**Pull request by bardisg**
## Description of the change

Added:
- github actions to support the v1.1 CI pipeline and automate release process
- added jest configs to generate proper coverage of all source code
- moved analytics into src folder
- moved npm package files into packages/npm
- added sonarcloud config
- added new build env for moder only browsers

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

[> Fix [#1]() ](https://www.notion.so/rudderstacks/CI-CD-pipeline-JS-3d99ddf9b6f64501acb0cd567ced65ed)

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

- - -
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-sdk-js/729)

### CODE REVIEW

File: .github/workflows/notion-pr-sync.yml
Comment (saikumarrs) [inline .github/workflows/notion-pr-sync.yml:1]: Is this action still required?

File: package.json
Comment (saikumarrs) [inline package.json:153]: How is this info consumed into rollup, and the modern SDK is of a smaller size?

File: .github/workflows/notion-pr-sync.yml
Comment (bardisg) [reply .github/workflows/notion-pr-sync.yml:1]: Still trying to get confirmation from the engineering leads channel, will chase again

File: .github/pull_request_template.md
Comment (bardisg) [reply .github/pull_request_template.md:1]: We will do this but lets not block this PR for this one please

File: .github/workflows/slack-notify.yml
Comment (bardisg) [reply .github/workflows/slack-notify.yml:19]: I will move this to secret so we can change without a commit. Not dangerous to be exposed though

File: .size-limit.js
Comment (bardisg) [reply .size-limit.js:9]: Sure I will do this

File: jest.config.js
Comment (bardisg) [reply jest.config.js:31]: Will move the source around and auto fix linting issues in a subsequent PR

File: package.json
Comment (bardisg) [reply package.json:29]: All path black/white lists will be changed once I move all source code into relevant folders inside src in a separate PR

File: package.json
Comment (bardisg) [reply package.json:153]: Rollup uses babel to transpile and build the code output. Babel uses presets to define what it will transpile to. The preset env respects the browserlist consiguration based on selected browserlist env.

More tools respect browserlist in the same way see https://github.com/browserslist/browserslist top of the file

You can see what is the full list of browsers that will be supported be npm run check:support and test configurations at https://browsersl.ist/ right side of the page

File: .github/workflows/publish-new-release.yml
Comment (bardisg) [reply .github/workflows/publish-new-release.yml]: done

File: .github/workflows/slack-notify.yml
Comment (bardisg) [reply .github/workflows/slack-notify.yml]: done

File: .size-limit.js
Comment (bardisg) [reply .size-limit.js:9]: done

**Comment by bardisg** [comment]
size checks fails as the target branch dist folder structure is different now. no file size has been amended though. All will be fine after the merge

File: .github/workflows/notion-pr-sync.yml
Comment (saikumarrs) [reply .github/workflows/notion-pr-sync.yml:1]: @sivashanmukh, as per our discussion on a thread almost a month ago, we don't need this anymore, right?

Comment (saikumarrs) [reply .github/workflows/notion-pr-sync.yml:1]: Resolving for now to unblock the PR.

**Review by saikumarrs — APPROVED** [review]

**Pull request merged**
